File: handlerclaw/lib/core/data/dto/file_dto.dart
class FileDto {
  final String id;
  final String fileName;
  final String fileType;
  final int fileSize;
  final String filePath;

  FileDto({
    required this.id,
    required this.fileName,
    required this.fileType,
    required this.fileSize,
    required this.filePath,
  });

  factory FileDto.fromJson(Map<String, dynamic> json) {
    return FileDto(
      id: json['id'] ?? '',
      fileName: json['file_name'] ?? '',
      fileType: json['file_type'] ?? '',
      fileSize: json['file_size'] ?? 0,
      filePath: json['file_path'] ?? '',
    );
  }
}

File: api/src/modules/files/application/dtos/file.dto.ts
import type { IFile } from "../../domain/entities/file.entity.js";

export type UploadFileResponse = {
  file: IFile;
};

export type UploadFileRequest = {
  fileName: string;
  fileType: string;
  fileSize: number;
  filePath: string;
};

/**
 * Repository Data Contracts
 */
export type CreateFileData = {
  fileName: string;
  fileType: string;
  fileSize: number;
  filePath: string;
};
